### ISSUE TITLE
Just a few more Text/Paragraph Formatting Choices?

### ISSUE THREAD

**Issue by weavertheme** [WordPress/gutenberg#7162]
**Is your feature request related to a problem? Please describe.**
FIRST! GB 3.0 is really getting there. The insert new block, and the much better ability to select blocks with right/left alignment and in the Column block is way better! It seems to be getting really close. **Way to go!**

That said, I think the formatting choices allowed for simple text (paragraphs, text blocks) is just a bit under powered. Bold, Italic, Strickout, and Link is just not quite enough. True, many of the format buttons found on the standard TinyMCE editor are not needed because of other blocks, but some are just missing. People will complain they can't do some very basic formatting. I will list the buttons I think should be added (even if only for pl;ugin blocks.)

**Describe the solution you'd like**
Please consider re-adding:

- Underscore - many people like that and will miss it
- Full Justify alignment. With auto-hyphenate, it can look pretty good.
- Color for selected text - just for emphasis
- Special Characters - very important for technical articles

And maybe:

- Font families - a web-safe default set,  and a way for themes to provide an alternate set. This is likely a longer term option.
- Emoticons
- Theme settable insertion content - a pretty common ability used by many themes and plugins to insert shortcodes or other content into the middle of standard text. It would be very nice to add shortcodes to the middle of text - especially they are "live" shortcodes. Just dreaming for now, I suspect.

**Comment by tofumatt** [comment]
It appears text inside Paragraph blocks can be underlined with a keyboard shortcut, so its omission is just a UI thing. Seems like there's space for it, so that one seems no bother to add.

Using colour alone for text emphasis is not accessible, so I don't think should be encouraged. I'd say it's up to a theme to change colour for bold/strong text in that case.

Insertion of special characters is probably better left to the OS…

This issue covers a lot, but I'd say it's worth discussing just the addition of underline to the paragraph block. Thoughts, @karmatosed?

**Comment by maddisondesigns** [comment]
+1 for Special Characters!

It's incredibly useful having that in TinyMCE, for inserting characters like currency signs (¢€£¥), Copyright and Trademark symbols (©®™) and fractions and en/em dashes (¼½¾–—). I would be very surprised if many people knew the OS shortcuts for all those characters. I know I dont.

**Comment by weavertheme** [comment]
Pert of this is really a marketing issue. When Gutenberg becomes integrated, people are going to whine a lot about "I can't underline any more, or I can't insert special characters."  I don't deeply understand the interface to TinyMCE from the GB editor, but any of the menu editing options that are available directly should be considered as seems to be the case for underline.

I would expect that justified text and special characters could be there, too. 

But really, don't forget the marketing aspect of this. There is a lot of negative press all over the web about Gutenberg. I don't think I've ever seen another plugin with as many negative reviews as Gutenberg. And anything that isn't hard to do, like adding formatting buttons to the Paragraph and text input fields in other blocks will help people adjust. 

And no one wants to learn keyboard shortcuts like OD special character inputs. Probably 95% or more of all users don't even know they can even do things like that, let alone how to use them. But seeing an icon on the edit bar - that is easy. And that's the point - ease of use without unnecessary limitations.

**Comment by chrisvanpatten** [comment]
[Underline and Justify buttons haven't been in Core's default formatting bar since WP 4.7.](https://wptavern.com/wordpress-4-7-removes-the-underline-and-justify-buttons-from-the-editor)

A few related issues:

#6833 
#7131 
#2747

**Comment by karmatosed** [comment]
Thanks for this suggestion. I think it is absolutely important to think about what are the most used things and then take from there adding. A few of these have come up before.

We have a few options with these, we either add them into core or we see about adding through a plugin and seeing if people really need them.

Out of the list I think right now having an underscore could be useful, but I am not convinced it needs working on as a priority. The rest I would like to have more testing to see if they are something everyone wants or just a smaller section. That's not saying we don't expand to include that but we should avoid the kitchen sink.

**Comment by alicethedreamer** [comment]
I agree that Gutenberg editor needs text formatting options (special characters, unline, superscript, indent). With 4.0 now, Gutenberg makes it very fast to create pages, but I cannot depend on it alone if it misses these formating options, I hate to copy and paste from elsewhere, or add html codes just to add superscript. The toolbar can be made extendable with more options, depending on user preferences. I have been searching for a plugin for these missing text formatting options to no avail. I understand that the developers want to keep Gutenberg tidy and non-cluttered, but an editor has to have these formatting options to be taken seriously, IMHO.
